app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,187 @@
+module ManageIQ::Providers::Amazon::AgentCoordinatorWorker::Runner::ResponseThread
+  require 'xml/xml_utils'
+  require 'scanning_mixin'
+  include ScanningMixin
+
+  #
+  # This thread startup method is called by the AgentCoordinatorWorker::Runner
+  # at the end of the do_before_work_loop method.
+  #
+  def start_response_thread
+    Thread.new do
+      _log.info("Starting AWS SSA Response Thread")
+      begin
+        # Wait a bit for the Agent Coordinators to be started
+        sleep(20)
+        response_handler_loop
+      rescue SystemExit
+        _log.info("SystemExit received, exiting AWS SSA Response Thread")
+        stop_response_thread
+        Thread.exit
+      rescue => err
+        _log.error("start_response_thread: Error [#{err}]")
+        _log.error("start_response_thread: Error - [#{err.backtrace.join("\n")}]")
+      end
+    end
+  end
+
+  def stop_response_thread
+    @shutdown_instance_wait_thread = true
+  end
+
+  def response_handler_loop
+    @shutdown_instance_wait_thread = nil
+    until @shutdown_instance_wait_thread
+      @coordinators.each do |coord|
+        _log.debug("Checking replies for Agent Coordinator Provider #{coord.ems.name}")
+        break if @shutdown_instance_wait_thread
+        if coord.reply_queue_empty?
+          _log.debug("No Replies visible for Provider #{coord.ems.name}")
+        else
+          ssaq_args = {}
+          ssaq_args[:ssa_bucket]    = coord.ssa_bucket
+          ssaq_args[:reply_queue]   = coord.reply_queue
+          ssaq_args[:request_queue] = coord.request_queue
+          ssaq_args[:region]        = coord.ems.provider_region
+          ssaq_args[:sqs]           = coord.sqs
+          ssaq_args[:s3]            = coord.s3
```

Comment:
Prefer a hash literal instead of a hash, then one-by-one adding keys...

```ruby
ssaq_args = {
  :ssa_bucket    => coord.ssa_bucket,
  :reply_queue   => coord.reply_queue,
  :request_queue => coord.request_queue
  ...
}
```